### ISSUE TITLE
Treesitter based "tagbar like" plugin

### ISSUE THREAD

**Issue by tuckerrc** [nvim-lua/wishlist#25]
**What?**
I would love to see a treesitter based [tagbar like](https://preservim.github.io/tagbar/) plugin. I think treesitter is perfect for this! 

**Why?**
I love tagbar but it requires a third party binary (ctags, universal-tags, etc.) to work.

**Potential existing implementations:**

 - [Tagbar](https://preservim.github.io/tagbar/)
 - [symbols-outline.nvim](https://github.com/simrat39/symbols-outline.nvim) (uses LSP instead of TS)

**Potential pitfalls*:*

Different languages may require different kinds of queries (I don't know enough about treesitter to know if this will be an issue). 

edit: Added another potential implementation

**Comment by disrupted** [comment]
agreed, there's [symbols-outline.nvim](https://github.com/simrat39/symbols-outline.nvim) but it uses LSP instead of Treesitter

**Comment by matu3ba** [comment]
This is afaik currently not feasible for bigger files due to performance issues inside nvim-treesitter.

Even if possible: Why introduce a third type of tagbar? universal-ctags are most efficient/fast for huge to big repos and the lsp can handle the rest.

Treesitter also cant resolve macros and alike without help of the lsp, so I dont exactly understand the argument and the use case.
